smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -357,6 +357,34 @@ contract MixinKeys is
     return key.tokenId;
   }
 
+  /**
+   * Merge existing keys
+   * @param _tokenIdFrom the id of the token to substract time from
+   * @param _tokenIdTo the id of the destination token  to add time
+   * @param _amount the amount of time to transfer (in seconds)
+   */
+  function mergeKeys(
+    uint _tokenIdFrom, 
+    uint _tokenIdTo, 
+    uint _amount
+    ) public {
+
+    // checks
+    _isKey(_tokenIdFrom);
+    _isValidKey(_tokenIdFrom);
+    _onlyKeyManagerOrApproved(_tokenIdFrom);
+    _isKey(_tokenIdTo);
+    
+    // make sure there is enough time remaining
+    require(keyExpirationTimestampFor(_tokenIdFrom) - block.timestamp > _amount, 'NOT_ENOUGH_TIME');
```

Comment:
Should we also make sure there's enough to cover for the transfer fee (when applicable)?

Reply:
oh I thought the transfer fee was just for `shareKey`. Will add here then